helpers/prism.js

```diff
@@ -0,0 +1,432 @@
+	.token.prolog,
+	.token.doctype,
+	.token.cdata {
+		color: var(--d2l-code-token-comment);
+	}
+
+	.token.punctuation { color: var(--d2l-code-token-punctuation); }
+	.token.number { color: var(--d2l-code-token-number); }
+	.token.operator { color: var(--d2l-code-token-operator); }
+
+	.token.interpolation { color: var(--d2l-code-token-interpolation); }
+	.token.attr-name { color: var(--d2l-code-token-attribute-name); }
+	.token.constant { color: var(--d2l-code-token-constant); }
+	.token.property { color: var(--d2l-code-token-property); }
+
+	.token.tag { color: var(--d2l-code-token-tag); }
+	.token.boolean { color: var(--d2l-code-token-boolean); }
+	.token.entity { color: var(--d2l-code-token-entity); }
+	.token.interpolation-punctuation { color: var(--d2l-code-token-interpolation-punctuation); }
+
+	.token.function { color: var(--d2l-code-token-function); }
+
+	.token.class-name { color: var(--d2l-code-token-class-name); }
+	.token.symbol { color: var(--d2l-code-token-symbol); }
+
+	.token.builtin { color: var(--d2l-code-token-builtin); }
+	.token.keyword { color: var(--d2l-code-token-keyword); }
+	.token.atrule { color: var(--d2l-code-token-atrule); }
+
+	.token.selector { color: var(--d2l-code-token-selector); }
+
+	.token.important { color: var(--d2l-code-token-important); }
+	.token.regex { color: var(--d2l-code-token-regex); }
+
+	.token.string { color: var(--d2l-code-token-string); }
+	.token.char { color: var(--d2l-code-token-char); }
+	.token.attr-value { color: var(--d2l-code-token-attribute-value); }
+
+	.token.url { color: var(--d2l-code-token-url); }
+
+	.language-css .token.string,
+	.style .token.string,
+	.token.entity,
+	.token.operator,
+	.token.url {
+		background-color: transparent;
+	}
+
+	pre[class*="language-"].line-numbers {
+		counter-reset: linenumber;
+		padding-left: 3.8em;
+		position: relative;
+	}
+
+	pre[class*="language-"].line-numbers > code {
+		position: relative;
+		white-space: inherit;
+	}
+
+	.line-numbers .line-numbers-rows {
+		border-right: 1px solid var(--d2l-code-line-numbers-separator);
+		font-size: 100%;
+		left: -3.8em;
+		letter-spacing: -1px;
+		pointer-events: none;
+		position: absolute;
+		top: 0;
+		-webkit-user-select: none;
+		-moz-user-select: none;
+		-ms-user-select: none;
+		user-select: none;
+		width: 3em; /* works for line-numbers below 1000 lines */
+	}
+
+	.line-numbers-rows > span {
+		counter-increment: linenumber;
+		display: block;
+	}
+
+	.line-numbers-rows > span::before {
+		color: var(--d2l-code-line-numbers);
+		content: counter(linenumber);
+		display: block;
+		padding-right: 0.8em;
+		text-align: right;
+	}
+`;
+
+const getLanguageInfo = elem => {
+	const classes = elem.classList;
+	for (let i = 0; i < classes.length; i++) {
+		if (classes[i].startsWith('language-')) {
+			const key = classes[i].substring(9);
+			const desc = codeLanguages.get(key);
+			if (desc) return { key: key, desc: desc };
+		}
+	}
+	return { key: 'plain', desc: codeLanguages.get('plain') };
```

Comment:
If the specified language is not recognized, we fall back to the `plain` language component which will be formatted without syntax highlighting, but will still include line numbers if requested.